backend/src/yeomanui.ts

```diff
@@ -359,23 +360,10 @@ export class YeomanUI {
     this.youiEvents.doGeneratorDone(false, errorMessage);
   }
 
-  private setGenInstall(gen: any, generatorName: string) {
-    const originalPrototype = Object.getPrototypeOf(gen);
-    const originalGenInstall = _.get(originalPrototype, "install");
-    if (originalGenInstall && !originalPrototype._uiInstall) {
-      originalPrototype._uiInstall = true;
-      originalPrototype.install = async () => {
-        try {
-          this.youiEvents.doGeneratorInstall();
-          await originalGenInstall.call(gen);
-        } catch (error) {
-          this.onGeneratorFailure(generatorName, error);
-        } finally {
-          originalPrototype.install = originalGenInstall;
-          delete originalPrototype._uiInstall;
-        }
-      };
-    }
+  private onGenInstall(gen: any) {
+    gen.on("method:install", () => {
+		this.youiEvents.doGeneratorInstall();
+	});
```

Comment:
Isn't the gen.on being called after the install method completed ?

Reply:
When it is completed another event is called. It's name is:  "done$<gen namespace>#install and it is fired by environment, not by a generator